diff --git a/Projects/Android/app/CMakeLists.txt b/Projects/Android/app/CMakeLists.txt
index e9db491b..363d0a57 100644
--- a/Projects/Android/app/CMakeLists.txt
+++ b/Projects/Android/app/CMakeLists.txt
@@ -11,10 +11,11 @@ set_source_files_properties("${ANDROID_NDK}/sources/android/cpufeatures/cpu-feat
 add_definitions("-DJUCE_ANDROID=1" "-DJUCE_ANDROID_API_VERSION=21" "-DJUCE_ANDROID_GL_ES_VERSION_3_0=1" "-DJUCER_ANDROIDSTUDIO_6AB8437A=1" "-DJUCE_APP_VERSION=3.10" "-DJUCE_APP_VERSION_HEX=0x30a00")
 
 include_directories( AFTER
-    "../../../ThirdParty/JUCE/modules/juce_audio_processors/format_types/VST3_SDK"
+    #"../../../ThirdParty/JUCE/modules/juce_audio_processors/format_types/VST3_SDK"
     "../../Projucer/JuceLibraryCode"
     "../../../ThirdParty/JUCE/modules"
     "${ANDROID_NDK}/sources/android/cpufeatures"
+    "../../../../aap-juce/modules"
 )
 
 enable_language(ASM)
@@ -24,7 +25,7 @@ if(JUCE_BUILD_CONFIGURATION MATCHES "DEBUG")
     include_directories( AFTER
         "../../Projucer/JuceLibraryCode"
         "../../../ThirdParty/JUCE/modules"
-        "../../../ThirdParty/JUCE/modules/juce_audio_processors/format_types/VST3_SDK"
+        #"../../../ThirdParty/JUCE/modules/juce_audio_processors/format_types/VST3_SDK"
         "../../../ThirdParty/ASIO/common"
         "../../../Source"
         "../../../Source/Core"
@@ -188,8 +189,10 @@ add_library( ${BINARY_NAME}
     "../../Projucer/JuceLibraryCode/include_juce_opengl.cpp"
     "../../Projucer/JuceLibraryCode/JuceHeader.h"
     "../../Projucer/JuceLibraryCode/UnityBuild.cpp"
+    "../../../../../juceaap.cpp"
 )
 
+
 set_source_files_properties("../../Projucer/JuceLibraryCode/AppConfig.h" PROPERTIES HEADER_FILE_ONLY TRUE)
 set_source_files_properties("../../Projucer/JuceLibraryCode/BinaryData.h" PROPERTIES HEADER_FILE_ONLY TRUE)
 set_source_files_properties("../../Projucer/JuceLibraryCode/JuceHeader.h" PROPERTIES HEADER_FILE_ONLY TRUE)
@@ -209,6 +212,13 @@ find_library(android "android")
 find_library(glesv3 "GLESv3")
 find_library(egl "EGL")
 
+find_package(androidaudioplugin REQUIRED CONFIG)
+target_include_directories( ${BINARY_NAME} PRIVATE androidaudioplugin)
+target_compile_definitions( ${BINARY_NAME} PRIVATE
+        JUCEAAP_USE_UNIQUE_ID=1
+        JUCEAAP_HOSTED_PARAMETER=1
+)
+
 target_link_libraries( ${BINARY_NAME}
 
     ${log}
@@ -216,4 +226,5 @@ target_link_libraries( ${BINARY_NAME}
     ${glesv3}
     ${egl}
     "cpufeatures"
+    androidaudioplugin::androidaudioplugin
 )
diff --git a/Projects/Android/app/build.gradle b/Projects/Android/app/build.gradle
index cd1afcaf..c8554ea6 100644
--- a/Projects/Android/app/build.gradle
+++ b/Projects/Android/app/build.gradle
@@ -1,31 +1,24 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 28
+    namespace "org.androidaudioplugin.ports.juce.helio"
+    compileSdk = 33
     externalNativeBuild {
         cmake {
             path "CMakeLists.txt"
         }
     }
-    signingConfigs {
-        juceSigning {
-            storeFile     file("${System.properties['user.home']}${File.separator}helio.keystore")
-            storePassword "android"
-            keyAlias      "release"
-            keyPassword   "android"
-            storeType     "jks"
-        }
-    }
 
     defaultConfig {
         applicationId "fm.helio"
-        minSdkVersion    21
-        targetSdkVersion 28
+        minSdkVersion    29
+        targetSdkVersion 33
         externalNativeBuild {
             cmake {
-                arguments "-DANDROID_TOOLCHAIN=clang", "-DANDROID_PLATFORM=android-21", "-DANDROID_STL=c++_static", "-DANDROID_CPP_FEATURES=exceptions rtti", "-DANDROID_ARM_MODE=arm", "-DANDROID_ARM_NEON=TRUE", "-DCMAKE_CXX_STANDARD=14", "-DCMAKE_CXX_EXTENSIONS=OFF"
+                arguments "-DANDROID_TOOLCHAIN=clang", "-DANDROID_PLATFORM=android-21", "-DANDROID_STL=c++_shared", "-DANDROID_CPP_FEATURES=exceptions rtti", "-DANDROID_ARM_MODE=arm", "-DANDROID_ARM_NEON=TRUE", "-DCMAKE_CXX_STANDARD=17", "-DCMAKE_CXX_EXTENSIONS=OFF"
             }
         }
+        proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
     }
 
     buildTypes {
@@ -33,22 +26,17 @@ android {
              initWith debug
              debuggable    true
              jniDebuggable true
-             signingConfig signingConfigs.juceSigning
          }
          release {
              initWith release
              debuggable    false
              jniDebuggable false
-             signingConfig signingConfigs.juceSigning
          }
     }
 
     flavorDimensions "default"
     productFlavors {
         debug_ {
-            ndk {
-                abiFilters "armeabi-v7a", "arm64-v8a"
-            }
             externalNativeBuild {
                 cmake {
                     arguments "-DJUCE_BUILD_CONFIGURATION=DEBUG", "-DCMAKE_CXX_FLAGS_DEBUG=-O0", "-DCMAKE_C_FLAGS_DEBUG=-O0"
@@ -58,9 +46,6 @@ android {
             dimension "default"
        }
         release_ {
-            ndk {
-                abiFilters "armeabi-v7a", "arm64-v8a"
-            }
             externalNativeBuild {
                 cmake {
                     arguments "-DJUCE_BUILD_CONFIGURATION=RELEASE", "-DCMAKE_CXX_FLAGS_RELEASE=-O3", "-DCMAKE_C_FLAGS_RELEASE=-O3"
@@ -86,7 +71,8 @@ android {
     sourceSets {
         main.java.srcDirs +=
             ["../../../ThirdParty/JUCE/modules/juce_core/native/javacore/init",
-             "../../../ThirdParty/JUCE/modules/juce_core/native/javacore/app"]
+             "../../../ThirdParty/JUCE/modules/juce_core/native/javacore/app",
+             "../../../../aap-juce/java"]
 
         main.res.srcDirs +=
             []
@@ -96,9 +82,15 @@ android {
     }
 
     dependencies {
+        implementation 'androidx.startup:startup-runtime:1.1.1'
+        implementation 'org.androidaudioplugin:androidaudioplugin:0.7.5-pre'
+    }
+
+    buildFeatures {
+        prefab true
     }
 
-ndkVersion "21.3.6528147"
+ndkVersion "25.1.8937393"
 
 }
 
diff --git a/Projects/Android/app/src/main/AndroidManifest.xml b/Projects/Android/app/src/main/AndroidManifest.xml
index f6967cb9..dca70122 100644
--- a/Projects/Android/app/src/main/AndroidManifest.xml
+++ b/Projects/Android/app/src/main/AndroidManifest.xml
@@ -1,11 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
 
-<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="310" android:versionName="3.10"
-          package="fm.helio">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:versionCode="310" android:versionName="3.10">
+  <queries>
+    <intent>
+      <action android:name="org.androidaudioplugin.AudioPluginService.V2" />
+    </intent>
+  </queries>
   <supports-screens android:smallScreens="true" android:normalScreens="true" android:largeScreens="true" android:anyDensity="true"
                     android:xlargeScreens="true"/>
   <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
   <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
+  <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
   <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
   <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
   <uses-permission android:name="android.permission.BLUETOOTH"/>
@@ -14,11 +21,19 @@
   <application android:label="@string/app_name" android:name="com.rmsl.juce.JuceApp" android:theme="@android:style/Theme.Material.NoActionBar.Fullscreen"
                android:icon="@drawable/icon" android:hardwareAccelerated="false">
     <activity android:name="android.app.Activity" android:label="@string/app_name" android:configChanges="keyboardHidden|orientation|screenSize"
-              android:screenOrientation="userLandscape" android:launchMode="singleTask" android:hardwareAccelerated="true">
+              android:screenOrientation="userLandscape" android:launchMode="singleTask" android:hardwareAccelerated="true" android:exported="true">
       <intent-filter>
         <action android:name="android.intent.action.MAIN"/>
         <category android:name="android.intent.category.LAUNCHER"/>
       </intent-filter>
     </activity>
-  </application>
+    <provider android:name="androidx.startup.InitializationProvider"
+        android:authorities="${applicationId}.juce_app_initializer"
+        android:exported="false"
+        tools:node="merge"
+        tools:replace="android:authorities">
+      <meta-data android:name="org.androidaudioplugin.hosting.AudioPluginClientInitializer" android:value="androidx.startup"/>
+      <meta-data android:name="org.androidaudioplugin.juce.JuceAppInitializer" android:value="androidx.startup" />
+    </provider>
+</application>
 </manifest>
diff --git a/Projects/Android/build.gradle b/Projects/Android/build.gradle
index 0fe5b4aa..797a11ea 100644
--- a/Projects/Android/build.gradle
+++ b/Projects/Android/build.gradle
@@ -1,16 +1,18 @@
 buildscript {
    repositories {
        google()
-       jcenter()
+       mavenCentral()
+       mavenLocal()
    }
    dependencies {
-       classpath 'com.android.tools.build:gradle:4.0.0'
+       classpath 'com.android.tools.build:gradle:7.3.1'
    }
 }
 
 allprojects {
    repositories {
        google()
-       jcenter()
+       mavenCentral()
+       mavenLocal()
    }
 }
diff --git a/Projects/Android/gradle/wrapper/gradle-wrapper.properties b/Projects/Android/gradle/wrapper/gradle-wrapper.properties
index 49479605..f26fd695 100644
--- a/Projects/Android/gradle/wrapper/gradle-wrapper.properties
+++ b/Projects/Android/gradle/wrapper/gradle-wrapper.properties
@@ -1 +1,5 @@
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip
\ No newline at end of file
+distributionBase=GRADLE_USER_HOME
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
+distributionPath=wrapper/dists
+zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
diff --git a/Projects/Projucer/Helio Workstation.jucer b/Projects/Projucer/Helio Workstation.jucer
index 71da5f79..9d0d369b 100644
--- a/Projects/Projucer/Helio Workstation.jucer	
+++ b/Projects/Projucer/Helio Workstation.jucer	
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <JUCERPROJECT id="lG607P" name="Helio" projectType="guiapp" version="3.10"
-              bundleIdentifier="fm.helio" includeBinaryInAppConfig="1" companyWebsite="https://helio.fm"
+              bundleIdentifier="org.androidaudioplugin.ports.juce.helio" includeBinaryInAppConfig="1" companyWebsite="https://github.com/atsushieno/aap-juce-helio"
               companyEmail="peter.rudenko@gmail.com" companyName="Peter Rudenko"
               displaySplashScreen="0" reportAppUsage="0" splashScreenColour="Dark"
               cppLanguageStandard="14" usePrecompiledHeaders="0" precompiledHeaderFileName=""
diff --git a/Projects/Projucer/JuceLibraryCode/AppConfig.h b/Projects/Projucer/JuceLibraryCode/AppConfig.h
index 159c5b51..898d583f 100644
--- a/Projects/Projucer/JuceLibraryCode/AppConfig.h
+++ b/Projects/Projucer/JuceLibraryCode/AppConfig.h
@@ -134,11 +134,11 @@
 // juce_audio_processors flags:
 
 #ifndef    JUCE_PLUGINHOST_VST
- #define   JUCE_PLUGINHOST_VST 1
+ #define   JUCE_PLUGINHOST_VST 0
 #endif
 
 #ifndef    JUCE_PLUGINHOST_VST3
- #define   JUCE_PLUGINHOST_VST3 1
+ #define   JUCE_PLUGINHOST_VST3 0
 #endif
 
 #ifndef    JUCE_PLUGINHOST_AU
diff --git a/Projects/Projucer/JuceLibraryCode/JuceHeader.h b/Projects/Projucer/JuceLibraryCode/JuceHeader.h
index 1e66e1fd..44c35164 100644
--- a/Projects/Projucer/JuceLibraryCode/JuceHeader.h
+++ b/Projects/Projucer/JuceLibraryCode/JuceHeader.h
@@ -34,7 +34,7 @@
      using the latest version of the Projucer or, if you aren't using the Projucer to manage your project,
      remove the JUCE_PROJUCER_VERSION define.
  */
- #error "This project was last saved using an outdated version of the Projucer! Re-save this project with the latest version to fix this error."
+ //#error "This project was last saved using an outdated version of the Projucer! Re-save this project with the latest version to fix this error."
 #endif
 
 #if ! DONT_SET_USING_JUCE_NAMESPACE
diff --git a/Source/Core/Audio/AudioCore.cpp b/Source/Core/Audio/AudioCore.cpp
index 13c3bb38..0a7cbf98 100644
--- a/Source/Core/Audio/AudioCore.cpp
+++ b/Source/Core/Audio/AudioCore.cpp
@@ -23,12 +23,17 @@
 #include "MetronomeSynthAudioPlugin.h"
 #include "SerializationKeys.h"
 #include "AudioMonitor.h"
+#include "aap/core/host/audio-plugin-host.h"
+#include "juceaap_audio_processors/juceaap_audio_plugin_format.cpp"
+
 
 void AudioCore::initAudioFormats(AudioPluginFormatManager &formatManager)
 {
     formatManager.addDefaultFormats();
     formatManager.addFormat(new InternalIODevicesPluginFormat());
     formatManager.addFormat(new BuiltInSynthsPluginFormat());
+    //aap::PluginClientSystem::getInstance()->getInstalledPlugins();
+    formatManager.addFormat(new juceaap::AndroidAudioPluginFormat());
 }
 
 AudioCore::AudioCore()
